Cache/IO.lean

```diff
@@ -406,4 +405,81 @@ def lookup (hashMap : ModuleHashMap) (paths : List FilePath) : IO Unit := do
       println! "  comment: {line}"
   if err then IO.Process.exit 1
 
+/--
+Parse a string as either a path or a Lean module name.
+TODO: If the argument describes a folder, use `walkDir` to find all `.lean` files within.
+
+Return tuples of the form ("module name", "path to .lean file").
+
+The input string `arg` takes one of the following forms:
+
+1. `Mathlib.Algebra.Fields.Basic`: there exists such a Lean file
+2. `Mathlib.Algebra.Fields`: no Lean file exists but a folder (TODO)
+3. `Mathlib/Algebra/Fields/Basic.lean`: the file exists (note potentially `\` on Windows)
+4. `Mathlib/Algebra/Fields/`: the folder exists (TODO)
+
+Not supported yet:
+
+5. `Aesop/Builder.lean`: the file does not exist, it's actually somewhere in `.lake`.
+
+Note: An argument like `Archive` is treated as module, not a path.
+-/
+def leanModulesFromSpec (sp : SearchPath) (argₛ : String) :
```

Comment:
Arguably this is ill-defined; is `Foo.Bar.lean` a reference to `./Foo.Bar.lean` or `Foo/Bar/lean.lean`?

Reply:
(maybe the answer is to allow a leading `./` to disambiguate)

Reply:
currently it's a reference to `./Foo.Bar.lean` especially since `Mathlib.lean` should be parsed that way. One add `./` but requiring it would not work for `Mathlib.lean`

Would it be desired to just try out both options with something like to following?

```lean
  if arg.extension == "lean" then
    let mut mods := #[]

    -- try if a file `Foo/Bar/lean.lean` exists
    let mod := argₛ.toName
    let sourceFile ← Lean.findLean sp mod
    if ← sourceFile.pathExists then
      mods := mods.append #[(mod, sourceFile)]

    -- try if a file `./Foo.Bar.lean` exists
    let mod₂ : Name := arg.withExtension "" |>.components.foldl .str .anonymous
    if (← arg.pathExists) then
      mods := mods.append #[(mod₂, arg)]
    return .ok mods
```

Reply:
> Arguably this is ill-defined; is `Foo.Bar.lean` a reference to `./Foo.Bar.lean` or `Foo/Bar/lean.lean`?

Alternatively, just assume that no one is going to do this. If they do, they can file a bug report, and we can tell them to stop being annoying. :-)

Reply:
At any rate we should make sure what we implement is consistent with whatever [lean4#2756](https://github.com/leanprover/lean4/pull/2756) decides.